### PR TITLE
Contributors page frontend update

### PR DESCRIPTION
**Pull request by nikkuAg**
Hey  **@chimosky**,

Issue: #335,

I have added a new contributors.css in css folder for the design of the contributor's page.

### CODE REVIEW

**Comment by nikkuAg** [comment]
Screenshot from 2021-12-31 18-49-20


I have made a design as attached. I have created a new CSS file and defined the css by giving class to various component of the profile.md file.

**Comment by quozl** [comment]
Appearance review; good layout, but why did you hide every entry behind a collapsible button?  That wasn't required, and if you think about it, all that does is hide contributors based on the alphabet.  Why would we want to do that?

Content review; the list of contributors is terribly out of date, and so it seems like a waste of time to change the appearance for bad data.

HTML5 review;

* the changes to the HTML and the new CSS file do not seem to use any of the existing Bootstrap styles, instead introducing a set of new styles; this is duplication of effort, and the page will look unlike the other pages,
* the added JavaScript seems hard to maintain, especially as it seems insufficiently factored,
* the stylesheet would have to be changed for _every new contributor_.

Hope that helps, sorry for the delay.

**Review by chimosky — CHANGES_REQUESTED** [review]

File: css/contributors.css [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+.link:visited, .link:active{
+    display: inline !important;
+    text-decoration: underline;
+    color: white;
+    cursor:pointer;
+}
+.link:hover{
+    color: black;
+    text-decoration: underline white;
+}
```
[the remark this answers is not in the document]
Comment (nikkuAg) [reply css/contributors.css:47]: Ok

**Comment by nikkuAg** [comment]
> Appearance review; good layout, but why did you hide every entry behind a collapsible button? That wasn't required, and if you think about it, all that does is hide contributors based on the alphabet. Why would we want to do that?
> 
> Content review; the list of contributors is terribly out of date, and so it seems like a waste of time to change the appearance for bad data.
> 
> HTML5 review;
> 
> * the changes to the HTML and the new CSS file do not seem to use any of the existing Bootstrap styles, instead introducing a set of new styles; this is duplication of effort, and the page will look unlike the other pages,
> * the added JavaScript seems hard to maintain, especially as it seems insufficiently factored,
> * the stylesheet would have to be changed for _every new contributor_.
> 
> Hope that helps, sorry for the delay.

Thanks for you review.
I will keep these points in mind and will incorporate those changes.

**Comment by chimosky** [comment]
In addition to what has been said, you might also want to checkout my earlier comment on the issue. https://github.com/sugarlabs/www-sugarlabs/issues/335#issuecomment-896097225

**Comment by nikkuAg** [comment]
> 

Hey @quozl,

I am facing a problem in applying the bootstrap. Class names like primary, m-2, p-2, etc. are default class names in bootstrap and are not working. So I want to know which bootstrap is in use for the previous page.

**Comment by quozl** [comment]
I've not used Bootstrap.  I presume you mean style names?  Where are they defined, and is that definition included by the HTML?

**Comment by nikkuAg** [comment]
> I've not used Bootstrap. I presume you mean style names? Where are they defined, and is that definition included by the HTML?

But in the previous comment, you mentioned that I have not used any existing Bootstrap styles.

**Comment by quozl** [comment]
> But in the previous comment, you mentioned that I have not used any existing Bootstrap styles.

None of your commit messages said that you used existing styles, and you've added a new CSS file just for the contributors page.  If I'm wrong in my assessment, please explain further.

In reviewing the commits again, I also note your addition of CDN for font-awesome.  We have tended to prefer all resources on the site rather than add external dependencies, though I see now that we have a few references to CDNs.

**Comment by nikkuAg** [comment]
> None of your commit messages said that you used existing styles, and you've added a new CSS file just for the contributors page. If I'm wrong in my assessment, please explain further.

You are right. But I am adding the new CSS file in such a way that even if you add new data then you don't have to add CSS

**Comment by nikkuAg** [comment]
Screenshot from 2022-01-07 19-48-23

I have changed the css and made this design.
I have to make a new file for CSS, but when we have to add new data we just have to include two class names whose reference could be found from the already written contributors data.
What I mean is that copy-paste the old contributor data and just change the details.

**Comment by quozl** [comment]
Thanks.  Why do names have a wavy underline?

You're embedding HTML in the markdown file profiles.md, is there a way to do it with Jekyll so you don't have to change a markdown file to mixed HTML?

**Comment by nikkuAg** [comment]
> Thanks.  Why do names have a wavy underline?

I have added wavy line to distinguish the name from other links available in the description field.

> You're embedding HTML in the markdown file profiles.md, is there a way to do it with Jekyll so you don't have to change a markdown file to mixed HTML?

Ok I will look into it and try to not change the markdown field

**Comment by srevinsaju** [comment]
> I am facing a problem in applying the bootstrap. Class names like primary, m-2, p-2, etc. are default class names in bootstrap and are not working.

@nikkuAg Just so that you are aware, the site is written in bootstrap 3, please check if the classes you have specified was already in the v3.3 (I couldn't find them, but I might be wrong). But I can confirm those are there in v4. Also, it would be better to use existing bootstrap classes than adding a new CSS file.

**Comment by nikkuAg** [comment]
> @nikkuAg Just so that you are aware, the site is written in bootstrap 3, please check if the classes you have specified was already in the v3.3 (I couldn't find them, but I might be wrong). But I can confirm those are there in v4. Also, it would be better to use existing bootstrap classes than adding a new CSS file.

Hey @srevinsaju,
Yes you are right that these classes are not in bootstrap v3.3. But of the fact that only using existing Bootstrap classes is not helpful as in my CSS file I have just added padding, margin, color, underline, which are not present in Bootstrap v3.3

**Comment by nikkuAg** [comment]
Hey @quozl @chimosky,

Can you please tell me if the design needs any more improvement or it is ready to merge

**Comment by quozl** [comment]
Thanks for asking.

On visual design.  The wavy underline looks odd to me, and I'm surprised it was chosen in order to distinguish from other links.  Nobody else has commented about it, so I don't know what the consensus is.  Other methods of distinguishing can be indentation or vertical space.  If you really want to use wavy underline then you will want to discover consensus, so we need responses from @srevinsaju and @chimosky on the wavy underline.  Their responses might be either (a) don't care, (b) thought Quozl said it all, (c) don't like it, or (d) like it.

On technical design.  I don't yet understand why the change cannot be done without adding HTML to the markdown file.  Perhaps @srevinsaju can suggest how to do it?  Sorry, but I haven't invested the time to find out how to do this (the `bundle install` failed on Ubuntu 22.04 beta).

**Comment by srevinsaju** [comment]
> The wavy underline looks odd to me

Yes, I agree with @quozl , The wavy underline looks off, at least for me, I kinda associate it as a wavy red underline shown by modern word processing tools to show grammatical or spelling errors. I prefer not using the wavy underline there.


I also do not agree with adding HTML to markdown. We have https://github.com/sugarlabs/www-sugarlabs/blob/master/_layouts/default.html which is being used by `profiles.md` if I understood correctly. One thing we could do is to have a custom `profile.html` which inherits from `default.html`. Doing so, we will be able to separate styling, and layouts from the data. I would like to keep the `profiles.md` untouched. The goal would be to, write a template `_layouts/profiles.html` which can marshall the data from `profiles.md` into the styles and layouts as you have tried to do within markdown. 

Additional learning resources: https://jekyllrb.com/docs/layouts/

PS: I also noted this file, https://github.com/sugarlabs/www-sugarlabs/blob/master/_layouts/profile.html, but I am not sure where its being used. But I hope the idea is being conveyed.

**Comment by srevinsaju** [comment]
Also, quoting @quozl from https://github.com/sugarlabs/www-sugarlabs/pull/340#issuecomment-1003669015, 
> the list of contributors is terribly out of date

@nikkuAg Have you thought of any innovative idea on how we would be able to recognize all contributors, with least maintenance effort on the website, so that it reflects the most up-to-date information?

**Comment by chimosky** [comment]
> Thanks for asking.
> 
> On visual design. The wavy underline looks odd to me, and I'm surprised it was chosen in order to distinguish from other links. Nobody else has commented about it, so I don't know what the consensus is. Other methods of distinguishing can be indentation or vertical space. If you really want to use wavy underline then you will want to discover consensus, so we need responses from @srevinsaju and @chimosky on the wavy underline. Their responses might be either (a) don't care, (b) thought Quozl said it all, (c) don't like it, or (d) like it.

I agree, the wavy underline should be removed.

**Comment by nikkuAg** [comment]
> @nikkuAg Have you thought of any innovative idea on how we would be able to recognize all contributors, with least maintenance effort on the website, so that it reflects the most up-to-date information?

Hey @srevinsaju,

Right now I have not thought of any  idea for thatas I was only thinking for the design part. As this part completes I will think about that.

**Comment by nikkuAg** [comment]
> I also do not agree with adding HTML to markdown. We have https://github.com/sugarlabs/www-sugarlabs/blob/master/_layouts/default.html which is being used by `profiles.md` if I understood correctly. One thing we could do is to have a custom `profile.html` which inherits from `default.html`. Doing so, we will be able to separate styling, and layouts from the data. I would like to keep the `profiles.md` untouched. The goal would be to, write a template `_layouts/profiles.html` which can marshall the data from `profiles.md` into the styles and layouts as you have tried to do within markdown.

Okay. I will reomve the wavy underline and apply the changes with altering `profiles.md` file

**Comment by nikkuAg** [comment]
Screenshot from 2022-01-11 22-13-56

Hey @quozl @srevinsaju 
I have changed profiles.md file and made a new profiles.html for adding some HTML and other types of CSS I have added using selectors.

**Comment by quozl** [comment]
Thanks for this work.  I approve the visual design.  On technical design,

* [ ] you again have missing end of line mark on a new file, `profiles.html`.  Please see if you can figure out what is in your workflow or tools that causes this.  In future please (a) look at the "`git diff`" and understand it before "`git commit`", and (b) look at the "Files changed" tab in the pull request to appreciate your change so far.
* [ ] in existing files `_layouts/*.html` a`{{ site.baseurl }}` is used as a prefix before stylesheets, should you use the same?
* [ ] some of your changes to `profiles.md` show errors; Bert has lost end of paragraph, Tony has blanks added to end of line, Walter has paragraphs unnecessarily rewrapped, Samson has broken links; please (a) look very carefully at the "Files changed" tab to see these, line by line, and (b) fix them so that your commits overall only change what is needed to change.  You may find `git diff master` helpful for this.

**Comment by nikkuAg** [comment]
Ok I will look into all those things

**Review by nikkuAg — COMMENTED** [review]
The changes in the profile.md file are only those that are required for styling and there is no HTML written.

**Comment by nikkuAg** [comment]
> Thanks for this work. I approve the visual design. On technical design

Hey @quozl,

I have updated all the changes you mentioned

**Comment by quozl** [comment]
Thanks.  I saw some more unrelated changes, and pushed a commit to fix them.  Please review 0a45d9a.

**Review by nikkuAg — COMMENTED** [review]
Changes reviewed and found that class names were not properly stated so updated those changes.

**Pull request merged**